3DAmsterdam/Assets/Netherlands3D/Scripts/Interface/Measuring/MeasuringLine.cs

```diff
@@ -0,0 +1,271 @@
+
+		ResetLine(); //Start hidden, wait for clicks
+		TakeInteractionPriority();
+
+		Selector.Instance.registeredClickInput.AddListener(PlacePoint);
+	}
+
+	public override void OnDisable()
+	{
+		base.OnDisable();
+		ResetLine();
+		Selector.Instance.registeredClickInput.RemoveListener(PlacePoint);
+	}
+
+	public override void Escape()
+	{
+		ResetLine();		
+		gameObject.SetActive(false);
+	}
+
+	private void DisplayPoint(Transform targetPoint)
+	{
+		targetPoint.gameObject.SetActive(true);
+	}
+
+	private void ResetLine()
+	{
+		placementStepIndex = -1;
+		lineRenderer.enabled = false;
+		lineRenderer.material = lineMaterial;
+		foreach (var linePoint in linePoints) linePoint.gameObject.SetActive(false);
+
+		if (distanceText) Destroy(distanceText.gameObject);
+	}
+
+	private void PlacePoint()
+	{
+		Vector3 placementPoint = (Selector.doingMultiselect) ? linePoints[1].transform.position : previewTargetPoint.position;
+		StepThroughPlacement(placementPoint);
+	}
+
+	private void PreviewNextPoint()
+	{
+		Vector3 previewPoint = CameraModeChanger.Instance.CurrentCameraControls.GetMousePositionInWorld();
+		PrepareColliders(previewPoint);
+
+		//Shoot ray to get precise point on collider
+		if (Physics.Raycast(Selector.mainSelectorRay, out RaycastHit hit))
+		{
+			var hitCollider = hit.collider.GetComponent<MeshCollider>();
+			if(hitCollider != targetCollider)
+			{
+				targetCollider = hitCollider;
+				targetMesh = targetCollider.sharedMesh;
+			}
+			previewPoint = hit.point;
+		}
+
+		//Move the next point in line to the previewposition if it was not the last point
+		if (placementStepIndex == 0)
+		{
+			if (Selector.doingMultiselect)
+			{
+				//Only do height, straight up when using modifier
+				linePoints[1].transform.position = new Vector3(linePoints[0].transform.position.x, previewPoint.y, linePoints[0].transform.position.z);
+
+				foreach (var linePoint in linePoints) linePoint.ChangeShape(MeasurePoint.Shape.HEIGHT);
+			}
+			else{
+				linePoints[1].transform.position = previewPoint;
+
+				foreach (var linePoint in linePoints) linePoint.ChangeShape(MeasurePoint.Shape.POINT);
+			}
+		}
+
+		previewTargetPoint.position = previewPoint;
+		AutoScalePointByDistance(previewTargetPoint);
+	}
+
+	private void PrepareColliders(Vector3 placementPoint)
+	{
+		//Make sure our targetlayers have colliders
+		foreach (var hitTargetLayer in targetLayers)
+		{
+			//Make sure all tiles close to mouse have a collider
+			hitTargetLayer.AddMeshColliders(placementPoint);
+		}
+	}
+
+	/// <summary>
+	/// Placement steps where 0 is clear/nothing shown, and the others place the points, and show the line
+	/// </summary>
+	/// <param name="placementPoint"></param>
+	private void StepThroughPlacement(Vector3 placementPoint)
+	{
+		placementStepIndex++;
+		if (placementStepIndex == linePoints.Count) placementStepIndex = 0;
+
+		if (placementStepIndex == 0)
```

Comment:
kan placementStepIndex -1 zijn? want de functie begint met placementStepIndex++

Reply:
yo bart; ja dat is bij het starten van de tool, dat is het geval wanneer er nog geen punten geplaatst zijn.
daarna heb je altijd een of twee punten geplaatst,totdat je de tool weer open.